x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authc/AuthenticationTestHelper.java

```diff
@@ -331,6 +342,8 @@ public Authentication build(boolean runAsIfNotAlready) {
                         if (user == null) {
                             user = randomUser();
                         }
+                        // User associated to API key authentication has empty roles
+                        user = stripRoles(user);
```

Comment:
Another bug for the test helper

Reply:
I'm Okay with the strip but I would've preferred an assert for the empty role name list, wdyt?

Reply:
I chose to strip the roles for usability reasons. It is challenging to randomize authentication freely if we assert the role list must be empty and the Authentication happens to be an API key (it can either API key or a token created by an API key or run-as by an API key). Since this is after all test code, I took the easier approach to strip roles. 

That said, stripping roles is in fact more akin to what happens in production code:
https://github.com/elastic/elasticsearch/blob/757f5c0795e1567b89783536ec1798b4f2380c03/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java#L722

which essentially also strips the roles.